### ISSUE TITLE
Accessibility:  Need a tabbable title in modal dialog

### ISSUE THREAD

**Issue by 508it** [infor-design/enterprise#2026]
**Describe the bug**
Keyboard focus should move to the title when the modal dialog is loaded so the title is read.

**To Reproduce In Landmark**
1. Go to https://hcm-tampm01-prd.inforcloudsuite.com/hcm/EmployeeSelfService/form/Employee%287004,700417%29.LRCFullEmployeeProfile?csk.showusingxi=true&csk.JobBoard=INTERNAL&csk.HROrganization=7004&menu=LRCEmployeeMenu.MyProfile
2. Click on any magnifying glass icon to launch a modal dialog
3. Tab until visual focus is in the modal dialog 
4. Visual focus only appears on toolbar in modal dialog after user tabs through calling page

**To Fix In EP**
1. http://master-enterprise.demo.design.infor.com/components/modal/example-index.html
2. Ensure the title is read by the screen reader. Possibly this means making it the thing that gets focus

**Expected behavior**
When user tabs, visual focus should start on the toolbar in the modal dialog

**See also**
https://jira.lawson.com/browse/LMCLIENT-24204

**Comment by tmcconechy** [comment]
Not sure if you mean toolbar or the title being read. I cant actually get in to any of the links in Landmark so just guessing how it looks. So a screen shot would be helpful.

**Comment by 508it** [comment]
In this case, we are referring to the toolbar (see attached) 
<img width="437" alt="modal_dialog_toolbar" src="https://user-images.githubusercontent.com/48891350/56756579-10ed0180-6760-11e9-9270-f12644573e45.png">
in the modal dialog. The visual focus should start on the toolbar when the modal dialog is opened.  The component example has a simple form and not a grid or toolbar, but focus is clearly on the first focusable element after it reads the labelled-by and described-by elements.

**Comment by tmcconechy** [comment]
Ok, this is a custom dialog and not something we have as a component. But do you think we should focus "Select Vereran Type" and then they could focus to that custom toolbar with a second tab?

**Comment by 508it** [comment]
Focus should most likely go to the first actionable element -  the toolbar.  The H1, Select Veteran Type element should be the aria-labeledBy, which means giving it focus would read the text twice when the landmark team fixes that association.  

The more serious issue is that nothing on the form gets focus by default, so the focus remains on background page and it take several tabs to actually get on to the modal, which violates the modal having it's own separate, self contained, wrapping tab order.  

The [component library modal ](http://master-enterprise.demo.design.infor.com/components/modal/example-index.html) you suggested we retest in [LMCLIENT-24204](https://jira.lawson.com/browse/LMCLIENT-24204) correctly puts the focus on the first actionable element and wraps around through the elements when tabbing on the modal. 

The component could possibly take the id of the first element to focus as an argument.

On a separate note, the component incorrectly calls out an aria-describedBy with an id that points to a div that contains the entire form.  This will make the entire form read out when the modal is loaded which can be a bit annoying.  The aria-describedBy should be a descriptive paragraph describing the form, an .audible block giving additional context to AT only, or just not used at all.

Aside from this issue with the aria-describedBy, the problems we highlighted are mostly on the landmark side.

**Comment by tmcconechy** [comment]
We can probably just set the modal to focus the first focusable element on the modal. This should fix this

This should have been working but maybe not for a case of a toolbar on the form, i think inputs will be focused if they are first. We usually dont recommend too much complexity on a modal, so toolbars arent that commonly used there.

**Comment by janahintal** [comment]
QA Passed. Will move to Done. Thanks
tested in http://master-enterprise.demo.design.infor.com/components/modal/test-toolbar.html across all browser, OS and devices
